### PR TITLE
[IMP] point_of_sale: add QR code in ticket screen to quickly retrieve orders.

### PR DESCRIPTION
**Pull request by shag-odoo**
Following this commit :
====
- QR code is been added to the ticket screen to retrieve orders. This will enhance the user experience and it will fasten the order searches in the ticket screen as well.

task- 4362235

### CODE REVIEW

**Comment by shag-odoo** [comment]
Hello @paso-odoo 
I have updated the code

Could you please have a look?
Thanks and have a wonderful day ahead :innocent:

**Review by paso-odoo — CHANGES_REQUESTED** [review]

File: addons/point_of_sale/static/src/app/screens/ticket_screen/ticket_screen.js [hunk 1]
```diff
@@ -112,6 +114,24 @@ export class TicketScreen extends Component {
             BACKSPACE,
         ]);
     }
+    onClickScanOrder(qrcode) {
+        if (qrcode) {
+            const ref = new URL(qrcode).searchParams.get("order_uuid");
+            const order = this.pos.models["pos.order"].getBy("uuid", ref);
```
Comment (caburj) [inline addons/point_of_sale/static/src/app/screens/ticket_screen/ticket_screen.js:120]: not enough, you should look at the server records as well. `onClickScanOrder` should be async to allow waiting for server call.

File: addons/point_of_sale/static/src/app/screens/ticket_screen/ticket_screen.js [hunk 2]
```diff
@@ -112,6 +114,24 @@ export class TicketScreen extends Component {
             BACKSPACE,
         ]);
     }
+    async onClickScanOrder(qrcode) {
+        if (qrcode) {
+            const ref = new URL(qrcode).searchParams.get("order_uuid");
+            const order = await this.pos.models["pos.order"].getBy("uuid", ref);
```
Comment (shag-odoo) [inline addons/point_of_sale/static/src/app/screens/ticket_screen/ticket_screen.js:120]: @caburj 
Did you mean this?
Please let me know if I misunderstood anything

Thanks in advance :+1:

Comment (caburj) [reply addons/point_of_sale/static/src/app/screens/ticket_screen/ticket_screen.js:120]: No. You'll have to do `this.pos.data.search` if `this.pos.models["pos.order"].getBy("uuid", ref)` didn't return anything.

Comment (shag-odoo) [reply addons/point_of_sale/static/src/app/screens/ticket_screen/ticket_screen.js]: I have updated the same
Could you please have a look.

**Comment by shag-odoo** [comment]
Hello @caburj 
Changes have been done
could you please have a look

Thanks and have a wonderful day ahead :innocent:

File: addons/point_of_sale/controllers/main.py [hunk 3]
```diff
@@ -137,6 +137,7 @@ def invoice_request_screen(self, **kwargs):
             'errors': errors,
             'banner_error': " ".join(errors.values()),
             'form_values': form_values,
+            'ref': order_uuid,
```
Comment (caburj) [inline addons/point_of_sale/controllers/main.py:140]: May I know what is this change for?

Comment (shag-odoo) [reply addons/point_of_sale/controllers/main.py:140]: yes it is no more required now
I will update it now

**Comment by caburj** [comment]
robodoo r+

**Pull request closed without merging**